### PR TITLE
Feature/api driven front end addition

### PR DESCRIPTION
**Pull request by TRoZZNL**
Contains the following changes:

- Added Content API to the Mediakiwi API
- Added Navigation API to the Mediakiwi API
- Added comments to API Methods
- Added comments to API Transport classes
- Replaced the obsolete and custom Rijndael with Rijndael Managed builtin for authentication
- Removed unused Swagger schemes
- Added LocalDevelopment options for Cookie and CORS policies
- Extended the UrlResolver class with additional properties, such as the current MediaKiwi User
- Added AdditionalResource class to Mediakiwi, for keeping track of all added resources

### CODE REVIEW

File: src/Sushi.Mediakiwi/Encryption/EncryptionHelper.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            if (s.Read(rawLength, 0, rawLength.Length) != rawLength.Length)
+            {
+                throw new SystemException("Stream did not contain properly formatted byte array");
+            }
+
+            byte[] buffer = new byte[BitConverter.ToInt32(rawLength, 0)];
+            if (s.Read(buffer, 0, buffer.Length) != buffer.Length)
+            {
+                throw new SystemException("Did not read byte array properly");
+            }
```
Comment (bart-van-der-drift) [inline src/Sushi.Mediakiwi/Encryption/EncryptionHelper.cs:140]: Why throw SystemExceptions? Is it maybe better to use InvalidOperationException?
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/using-standard-exception-types

File: src/Sushi.Mediakiwi/Framework/Api/MediakiwiJavascriptEvent.cs [hunk 2]
```diff
@@ -4,8 +4,8 @@ public enum MediakiwiJSEvent
     {
         none = 0,
         change = 1,
-        click = 1,        
-        blur = 2,
-        keyup = 3,
+        click = 2,        
+        blur = 3,
+        keyup = 4,
     }
```
Comment (bart-van-der-drift) [inline src/Sushi.Mediakiwi/Framework/Api/MediakiwiJavascriptEvent.cs:10]: Is there a reason these enum values do not start with an uppercase character?

**Review by bart-van-der-drift — COMMENTED** [review]
Please see comments

File: src/Sushi.Mediakiwi/Encryption/EncryptionHelper.cs (hunk 1 above)
Comment (TRoZZNL) [reply src/Sushi.Mediakiwi/Encryption/EncryptionHelper.cs:140]: Agreed, will throw an argument exception for the first and an InvalidOperationException for the second

File: src/Sushi.Mediakiwi/Framework/Api/MediakiwiJavascriptEvent.cs (hunk 2 above)
Comment (TRoZZNL) [reply src/Sushi.Mediakiwi/Framework/Api/MediakiwiJavascriptEvent.cs:10]: Not that i'm aware of, this was existing code albeit with a double value (1) which i fixed.
i'll check that the JSON output remains the same and will then rename them.

**Review by bart-van-der-drift — APPROVED** [review]

**Pull request merged**